app/helpers/application_notifications_helper.rb

```diff
@@ -0,0 +1,11 @@
+module ApplicationNotificationsHelper
+  # Translate the event of an activity to the path of a given notification
+  #
+  # @param [Course::Notification, UserNotification] notification whose view path is required
+  # @return [String] the path of a template
+  def notification_view_path(notification)
+    return unless notification.is_a?(UserNotification) || notification.is_a?(Course::Notification)
+    "/#{notification.activity.event.gsub('.', '/')}/#{notification.
+      class.name.underscore.sub('/', '_').pluralize}/#{notification.notification_type}"
```

Comment:
gsub?

You should split this line up into individual variables before using string interpolation to get the final result.